scripts/render-notebooks.R

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env Rscript
+#
+# Make rendered versions of .Rmd files with automagic citations with a specified
+# .bib file.
+
+# Example use:
+
+# Rscript scripts/render-notebooks.R \
+# -r 01-getting-started/getting-started.Rmd \
+# -b references.bib
+
+# Load library:
+library(optparse)
+
+# Set up optparse options
+option_list <- list(
+  make_option(
+    opt_str = c("-r", "--rmd"), type = "character",
+    default = NULL,
+    help = "File path to an .Rmd file that should have the bib_file added",
+    metavar = "character"
+  ),
+  make_option(
+    opt_str = c("-b", "--bib_file"), type = "character",
+    default = "references.bib", # Default is this file, but it can be changed
+    help = "File name of the references file. Can be any format pandoc works with. Will be normalized with normalizePath().",
+    metavar = "character"
+  )
+)
+
+# Parse options
+opt <- parse_args(OptionParser(option_list = option_list))
+
+# Make these file paths
+opt$rmd <- file.path(opt$rmd)
+
+# Check that the file exists
+if (!file.exists(opt$rmd)) {
+  stop("Rmd file specified with --rmd is not found.")
+}
+
+# Check that the file exists
+if (!file.exists(opt$bib_file)) {
+  stop("File specified for --bib_file option is not at the specified file path.")
+} else {
+  header_line <- paste("bibliography:", normalizePath(opt$bib_file))
+}
+
+# Specify the output file
+tmp_file <-  stringr::str_replace(opt$rmd, "\\.Rmd$", "-tmp-header-changed.Rmd")
+
+# Read in as lines
+lines <- readr::read_lines(opt$rmd)
+
+# Find which lines are the beginning and end of the header chunk
+header_range <- which(lines == "---")
+
+# Stop if no chunk found
+if (length(header_range) == 0) {
+  stop("Not finding the `---` which is at the beginning of the header.")
+}
+
+# Add the bibliography specification line at the beginning of the chunk
+new_lines <- append(lines, header_line, header_range[1])
+
+# Write to an outfile
+readr::write_lines(new_lines, tmp_file)
+
+# Specify final output file
+output_file <- stringr::str_replace(basename(opt$rmd), "\\.Rmd$", ".nb.html")
```

Comment:
I'm going to suggest you make the final output file an option too. Maybe use this if it isn't specified, but flexibility is good. (And move this up with the other option parsing.)

Don't use `basename()`, or things might not go where expected.
```suggestion
if (is.null(opt$html)){
  output_file <- stringr::str_replace(opt$rmd, "\\.Rmd$", ".html")
} else {
  output_file <- opt$html
}
```

Reply:
It would not let me specify the whole file path, only the file name. It seems like its assuming it will go in the same directory as the file I give it. 🤷‍♀️

Reply:
https://rmarkdown.rstudio.com/docs/reference/render.html says a path should work? maybe `normalizePath(opt$rmd)` here for safety?

Reply:
It seems it wants the file path relative to the `.Rmd` provided in input. Unless there's a way to not make it this way.

Reply:
normalizePath() works with this.

Reply:
That's why I think a full path may work. but don't stress about it if not. this is not a case we need to care about.

Reply:
It thinks my directory doesn't exist even though dir.exist() says it does.

Reply:
Updating my suggestion to include normalizePaths (requires the addition of an `--html` option)
```
if (is.null(opt$html)){
  output_file <- stringr::str_replace(normalizePath(opt$rmd), "\\.Rmd$", ".html")
} else {
  output_file <- normalizePath(opt$html)
}
```

Reply:
Implemented this option in [9a25c9e](https://github.com/AlexsLemonade/refinebio-examples/pull/150/commits/9a25c9e43203017bf3bbe67bb2eea76147d41d02)